Index: tagging/fields.py
===================================================================
--- tagging/fields.py	(revision 140)
+++ tagging/fields.py	(working copy)
@@ -3,7 +3,6 @@
 """
 from django.db.models import signals
 from django.db.models.fields import CharField
-from django.dispatch import dispatcher
 from django.utils.translation import ugettext_lazy as _
 
 from tagging import settings
@@ -30,7 +29,7 @@
         setattr(cls, self.name, self)
 
         # Save tags back to the database post-save
-        dispatcher.connect(self._save, signal=signals.post_save, sender=cls)
+        signals.post_save.connect(self._save, cls, True)
 
     def __get__(self, instance, owner=None):
         """
@@ -74,13 +73,13 @@
             value = value.lower()
         self._set_instance_tag_cache(instance, value)
 
-    def _save(self, signal, sender, instance):
+    def _save(self, **kwargs): #signal, sender, instance):
         """
         Save tags back to the database
         """
-        tags = self._get_instance_tag_cache(instance)
+        tags = self._get_instance_tag_cache(kwargs['instance'])
         if tags is not None:
-            Tag.objects.update_tags(instance, tags)
+            Tag.objects.update_tags(kwargs['instance'], tags)
 
     def __delete__(self, instance):
         """
Index: tagging/forms.py
===================================================================
--- tagging/forms.py	(revision 140)
+++ tagging/forms.py	(working copy)
@@ -1,7 +1,7 @@
 """
 Tagging components for Django's ``newforms`` form library.
 """
-from django import newforms as forms
+from django import forms
 from django.utils.translation import ugettext as _
 
 from tagging import settings
